unified_planning/test/test_protobuf_io.py

```diff
@@ -276,6 +276,13 @@ def test_validation_result(self):
 
                 self.assertEqual(validation_result, validation_result_up)
 
+    def test_temporal_hierarchical_goal(self):
```

Comment:
Great to have a complete example for mixed hierarchical and temporal.

It might be good to have a `test_timing` function that only test the different kind of timing expressions (start/end/global-start-global-end + 0/int/fraction).